File: back-end/app/infrastructure/persistence/models/experience_model.py
"""
Modelo SQLAlchemy para entradas de experiencia / trayectoria (no empleos inventados).
"""

from sqlalchemy import Column, String, Boolean, Integer, DateTime, Text, CHAR
from datetime import datetime
import uuid

from ..base import Base


class ExperienceModel(Base):
    """Filas de GET /portfolio/experience."""

    __tablename__ = "portfolio_experiences"

    id = Column(CHAR(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    title = Column(String(200), nullable=False)
    organization = Column(String(200), nullable=True)
    kind = Column(String(32), nullable=False)  # project | training | security
    location = Column(String(120), nullable=True)
    start_date = Column(String(7), nullable=False)  # YYYY-MM
    end_date = Column(String(7), nullable=True)
    current = Column(Boolean, default=False)
    summary = Column(Text, nullable=False)
    highlights = Column(Text)  # JSON list
    technologies = Column(Text)  # JSON list
    links = Column(Text)  # JSON object
    order = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, onupdate=datetime.utcnow)

File: front-end/portafolio/src/app/core/models/experience.model.ts
/** Alineado a ExperienceItemDTO (PR #48 / GET /portfolio/experience). */
export type ExperienceKind = 'project' | 'training' | 'security' | string;

export interface ExperienceLinks {
  github?: string | null;
  demo?: string | null;
}

export interface ExperienceItem {
  id: string;
  title: string;
  organization?: string | null;
  kind: ExperienceKind;
  location?: string | null;
  /** YYYY-MM */
  start_date: string;
  /** YYYY-MM o null */
  end_date: string | null;
  current: boolean;
  summary: string;
  highlights: string[];
  technologies: string[];
  links: ExperienceLinks;
  order: number;
}

export interface ExperienceListResponse {
  items: ExperienceItem[];
}

/** Alineado a CapabilitySkillDTO. */
export interface CapabilitySkill {
  name: string;
  category: string;
}

/** Alineado a CapabilitiesDTO (GET /portfolio/capabilities). */
export interface CapabilitiesResponse {
  roles: string[];
  skills: CapabilitySkill[];
}

/** Vista FE: chips derivados solo del endpoint de capabilities (no inventar stack). */
export interface CapabilityChip {
  label: string;
  kind: 'role' | 'skill';
  category?: string;
}
